futaba/delayed.py

```diff
@@ -40,20 +39,22 @@ def push(self, coro):
         self.queue.put_nowait(coro)
 
     async def main_loop(self):
-        while True:
+        for i in itertools.count():
             logger.debug("Waiting for new delayed event")
             coro = await self.queue.get()
-            await coro
 
-            logger.debug(
-                "Sleeping for %.3f seconds until next delayed event", self.delay
-            )
-            await asyncio.sleep(self.delay)
-
-    @property
-    def delay(self):
-        rate = self.queue.qsize() // self.bracket_size
-        return min(rate * self.rate, self.max_delay)
+            logger.debug("Got event, is #%d to be processed", i)
+            try:
+                await coro
+            except Exception as error:
+                logger.error("Error awaiting delayed event", exc_info=error)
```

Comment:
before, exceptions caused this to bail. did you mean to continue instead of reraise?

Reply:
This doesn't reraise anything, it just prints the exception to the log so we can see it failed.

Reply:
er, yeah, that's what I'm saying. before, exceptions stopped the loop. now, they don't. was that intentional?

Reply:
The loop should've never stopped, that was a bug. If the loop is dead then delayed events never get processed and just build up forever.